GrocifyApp.UI/src/components/modals/Products/ProductSectionModal.jsx

```diff
@@ -0,0 +1,85 @@
+import { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+
+//Internal components
+import BaseModal from '../BaseModal';
+import CustomInputApp from '../../CustomInputApp';
+import ProductSectionSelector from '../../Products/ProductSectionsSelector';
+import useApiRequest from '../../../hooks/useApiRequests';
+
+//Assets & Css
+import styles from './ProductSectionModal.module.scss';
+
+//Consts
+import { PlaceholderConsts, LabelConsts, ButtonConsts, ModalConsts } from '../../../consts/ENConsts';
+import InputType from '../../../consts/InputType';
+import ApiEndpoints from '../../../consts/ApiEndpoints';
+
+const ProductSectionModal = ({ onClose, onConfirm, section }) => {
+    const [isButtonDisabled, setButtonDisabled] = useState(true);
+    const [productSectionName, setProductSectionName] = useState(section != null ? section.name : '');
+    const [productSectionIcon, setProductSectionIcon] = useState(section != null ? section.icon : 'Home');
+    const { makeRequest } = useApiRequest();
+
+    useEffect(() => {
+        if (section) {
```

Comment:
please review, when editting, I'm able to remove section name and save it

Reply:
consider the following to be more easy to read:

`
const isSectionEdited = section && (productSectionName !== section.name || productSectionIcon !== section.icon);
const areInputsValid = productSectionName !== "" && productSectionIcon !== "";
        
setButtonDisabled(!areInputsValid || (section && !isSectionEdited));
`